aesd-char-driver/main.c

```diff
@@ -160,11 +171,19 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
         my_dev->cir_buff_size += len_rec; // Update circular buffer size
         write_buf.buffptr = my_dev->circular_buff;
         write_buf.size = my_dev->cir_buff_size;
+
```

Comment:
@Rishikesh38 I didn't catch this when reviewing assignment 8 - but the code at https://github.com/cu-ecen-aeld/assignments-3-and-later-Rishikesh38/pull/7/files#diff-e6d206de490054ca882e24fc06e3752cfc8240c67c2527207d6c56624f478901L121-L128 needs locking

Reply:
Yes, there might be a risk of losing a pointer to allocated memory without locking.